bitnami/rabbitmq/templates/_helpers.tpl

```diff
@@ -215,3 +205,19 @@ rabbitmq: memoryHighWatermark
       --set memoryHighWatermark.value="512MB"
 {{- end -}}
 {{- end -}}
+
+{{/*
+Validate values of rabbitmq - TLS configuration for Ingress
+*/}}
+{{- define "rabbitmq.validateValues.ingress.tls" -}}
+{{- if and .Values.ingress.enabled .Values.ingress.tls (not .Values.ingress.certManager) (not .Values.ingress.selfSigned) (empty .Values.ingress.extraTls) }}
+rabbitmq: ingress.tls
+    You enabled the TLS configuration for the default ingress hostname but
+    you did not enable any of the available mechanisms to create the TLS secret
+    to be used by the Ingress Controller.
+    Please use any of these alternatives:
+      - Use the `ingress.extraTls` and `ingress.secrets` parameters to provide your custom TLS certificates.
+      - Relay on cert-manager to create it by setting `ingress.certManager=true`
+      - Relay on Helm to create self-signed certificates by setting `ingress.selfSigned=true`
```

Comment:
So I think it's great that using cert manager is made easy (or still easy), given that's a large % of the use case. But IMO it'd be nicer if out of the other two options, providing your own `hostname-tls` cert was similarly easy. For that case, where I have used in the past:

```
ingress:
  tls: true
  hostname: localhost
```
and created my own `localhost-tls` secret in the same namespace (or the deployment fails), it was nice and simple, which is good for a common use-case. But now, even with this change, I think I'd need to do:

```
ingress:
  tls: true
  hostname: localhost
  extraTLS:
    - hosts: [localhost]
      secretName: localhost-tls
```
I think? Is that right? AFAICT, this complexity comes because we're trying to support out-of-the-box untrusted certs for a TLS setup. I'd be surprised if this is a large % use-case, in that, if I'm just wanting to test the chart, I won't use TLS. If I'm not testing but setting something up, I'll use TLS with a trusted cert. But I don't have a good grasp of the actual use-cases, so don't put much weight on my opinion here.

Reply:
> and created my own localhost-tls secret in the same namespace (or the deployment fails), it was nice and simple, which is good for a common use-case

IMHO, chart should be as self-contained as possible. In other words, I expect a chart to package everything I need to deploy XXX app. That's why I don't usually like these approaches when you actually to do two things: installing the chart + creating the secret.

> I think? Is that right?

Yes, you're completely right. That said, I don't think it's a high cost to pay. You just need to set a couple of parameters more but your use case is still covered.

Reply:
> IMHO, chart should be as self-contained as possible. In other words, I expect a chart to package everything I need to deploy XXX app.

But you fundamentally can't package the TLS secret you need to deploy XXX app. You can only have the package create an insecure TLS secret that cannot be used in a real deploy. So it prioritises the demo use-case over the real one, IMO.

> That's why I don't usually like these approaches when you actually need to do two things to deploy the app: installing the chart + creating the secret.

For a real deployment, you *need* to create the secret first and just want to specify the secret name, as per #6246, or use the default hostname-tls. For a demo deployment, you can still avoid two steps with your `selfSigned` option, I think.

Anyway, not my place to weigh in on this conversation, I'm just providing input as a user :) Happy for you and @javsalgar to figure out the best option here.

Reply:
> But you fundamentally can't package the TLS secret you need to deploy XXX app
> (...)
> For a real deployment, you need to create the secret first and just want to specify the secret name

That's not 100% accurate. You can also manage everything with the chart for the "real" deployment, see this example:

```yaml
ingress:
  enabled: true
  tls: true
  extraTls:
  - hosts:
    - rabbitmq.local
    secretName: FOO
  secrets:
  - name: FOO
    key: |-
      -----BEGIN RSA PRIVATE KEY-----
      ...
      -----END RSA PRIVATE KEY-----
    certificate: |-
      -----BEGIN CERTIFICATE-----
       ...
      -----END CERTIFICATE-----
 ```

> So it prioritises the demo use-case over the real one, IMO.
> (...)
> Anyway, not my place to weigh in on this conversation, I'm just providing input as a user :) Happy for you and @javsalgar to figure out the best option here.

I don't think it prioritizes one option or the another. It just gives freedom to the user different alternatives to manage the TLS termination.

Reply:
We've seen in several clients (for example those relying in solutions like Rancher) that they want to have everything set in the values, so something like 

```
  secrets:
  - name: FOO
    key: |-
      -----BEGIN RSA PRIVATE KEY-----
      ...
      -----END RSA PRIVATE KEY-----
    certificate: |-
      -----BEGIN CERTIFICATE-----
       ...
      -----END CERTIFICATE-----
```

Is becoming common and we were requested to include it. That's why we added some values like extraDeploy to our charts.

Reply:
> > For a real deployment, you need to create the secret first and just want to specify the secret name
> 
> That's not 100% accurate. You can also manage everything with the chart for the "real" deployment, see this example:

Yep, my mistake: you can manage the secret value via the chart, but you still have those two steps (as in, you need to create the actual cert first to have a value to include in the chart in that case). So I'm not sure what it changes: for any non-local-demo situation, you either have to create a cert and refer to it somehow (whether you pass the value to the chart or refer to the secret in the cluster doesn't change the two steps), or use something like cert-manager.

> > So it prioritises the demo use-case over the real one, IMO.
> > (...)
> > Anyway, not my place to weigh in on this conversation, I'm just providing input as a user :) Happy for you and @javsalgar to figure out the best option here.
> 
> I don't think it prioritizes one option or the another. It just gives freedom to the user different alternatives to manage the TLS termination.

It wasn't a big issue at all, but it did break the existing tls setup to provide auto-generated TLS certs (which, imo, should only be for demos).

> Is becoming common and we were requested to include it. That's why we added some values like extraDeploy to our charts.

Yeah, and that's great. My point was just about the TLS that got broken (again, not a big deal and easy to work around) when the chart was updated to enable auto-generated TLS certs. That is, as [above](https://github.com/bitnami/charts/pull/6235#discussion_r622629587), we were doing:

```
ingress:
  tls: true
  hostname: myhost.example.com
```
and ensuring our `myhost.example.com-tls` secret existed which worked and could not be simpler. Now it does not because the chart tries to create the `myhost.example.com-tls` which conflicts with the actual secret. But easy to update and work around it :)

Reply:
We'll go ahead with this approach since there's an easy way to workaround the problem for users like you used to manage the TLS with an external secret.
In any case, thanks so much for your priceless feedback! It's great to learn from other perspectives.